src/pushsource/_impl/backend/errata_source/errata_client.py

```diff
@@ -73,49 +107,197 @@ def get_raw_f(self, advisory_id):
         )
         return f_map(all_responses, lambda tup: ErrataRaw(*tup))
 
-    def _call_et(self, method_name, advisory_id):
-        method = getattr(self._errata_service, method_name)
-
+    def _call_et(self, method, advisory_id):
         # These APIs have had performance issues occasionally, so let's set up some
         # detailed structured logs which can be used to check the performance.
         LOG.debug(
             "Calling Errata Tool %s(%s)",
-            method_name,
+            method,
             advisory_id,
             extra={
                 "event": {
                     "type": "errata-tool-call-start",
-                    "method": method_name,
+                    "method": method,
                     "advisory": advisory_id,
                 }
             },
         )
         try:
-            out = method(advisory_id)
+            out = self._do_call(method, advisory_id)
+
             LOG.debug(
                 "Errata Tool completed call %s(%s)",
-                method_name,
+                method,
                 advisory_id,
                 extra={
                     "event": {
                         "type": "errata-tool-call-end",
-                        "method": method_name,
+                        "method": method,
                         "advisory": advisory_id,
                     }
                 },
             )
+
             return out
         except:
             LOG.debug(
                 "Failed to call Errata Tool %s(%s)",
-                method_name,
+                method,
                 advisory_id,
                 extra={
                     "event": {
                         "type": "errata-tool-call-fail",
-                        "method": method_name,
+                        "method": method,
                         "advisory": advisory_id,
                     }
                 },
             )
             raise
+
+
+class ErrataHTTPClient(ErrataClientBase):
+    def __init__(self, threads, url, keytab_path: str = None,
+                 principal: str = None, **retry_args):
+        super().__init__(threads, url, **retry_args)
+
+        self.get_advisory_data = partial(
+            self._call_et, "/api/v1/push_metadata/ftp_paths/{id}.json"
```

Comment:
I don't get this, why is get_advisory_data using ftp_paths?  Doesn't it need to use /api/v1/erratum/{id} like it did before this change?

Also, if this is a mistake, were tests somehow unable to catch it?  If so they should be improved.

Reply:
Bad copy paste job. I wanted to get opinions on this approach before going through and making all the tests work.